eq-author-api/src/validation/utils/index.js

```diff
@@ -0,0 +1,22 @@
+const getPath = path => {
+  if (typeof path !== "string") {
+    throw new Error("Path needs to be a string");
+  }
+  const dataPath = path.slice(1).split("/");
+
+  const current = {};
+  for (let i = 0; i < dataPath.length; i++) {
+    if (dataPath.length % 2 !== 0) {
+      current.field = dataPath[dataPath.length - 1];
+    }
+    if (!isNaN(dataPath[i])) {
+      current[dataPath[i - 1]] = dataPath[i];
+    }
+  }
```

Comment:
I like what you've done with the `getPath` function - good call refactoring all that other stuff out.

You can move the branch on non-even `dataPath.length` out of the loop though since it's not dependent upon the loop variable `i`. I tried this locally and it doesn't break anything:

```suggestion
  dataPath.forEach((segment, index) => {
    if(!isNaN(segment)) {
      current[dataPath[index - 1]] = segment;
    }
  });

  if(dataPath.length % 2 !== 0) {
    current.field = dataPath[dataPath.length - 1];
  }
```